### ISSUE TITLE
Markdown does not have syntax highlighting

### ISSUE THREAD

**Issue by jcowgar** [ki-editor/ki-editor#482]
The language is defined but no syntax highlighting exists.

**Comment by jcowgar** [comment]
May be related to #367.

**Comment by jcowgar** [comment]
Maybe it does have syntax highlighting, it is just very minimal. When changing themes (from Rose Pine Dawn) on some themes I see that meta data is highlighted as well as table characters. That seems to be it though.

<img width="785" alt="image" src="https://github.com/user-attachments/assets/45de4f93-832b-4047-b36d-8e23b26c323b" />

Is this expected for Markdown?

**Comment by wongjiahau** [comment]
For this I’m not sure, because I’m relying on the `highlight.scm`s of [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter/tree/master/queries) for syntax highlighting, while using the [themes from Zed editor](https://github.com/zed-industries/zed/tree/main/assets/themes).

**Comment by jcowgar** [comment]
Yes, that seems to be the issue. If I edit the cached highlights.scm and change:

```
(atx_heading
  (atx_h1_marker)) @markup.heading.1
```

to

```
(atx_heading
  (atx_h1_marker)) @function
```

then all H1's are now colorized.

I did look at Zed's `markdown/highlights.scm` file and put it in place of my local cached and it did not function either.

**Comment by wongjiahau** [comment]
Might be related: https://github.com/ki-editor/ki-editor/blob/master/src/themes/mod.rs#L419-L420

**Comment by wongjiahau** [comment]
We probably need to handle `markup.<XXX>` in this file https://github.com/ki-editor/ki-editor/blob/master/src/themes/from_zed_theme.rs#L87-L110

**Comment by jcowgar** [comment]
I added the most common Markup* values to from_zed_theme. Headers now contain syntax highlighting but nothing else really does. I ensured that I was not using an invalid selector, by setting them all to keyword (which is used for the Header) and nothing else highlighted.

I do see punctuation (for table) and list bullets have coloring.

So, maybe that is enough to call this fixed.

<img width="435" alt="image" src="https://github.com/user-attachments/assets/65c3e5f4-0023-4170-b983-222706f37330" />
